### PR TITLE
cdc: split the initializer from the endpoint

### PR DESCRIPTION
**Pull request by hi-rustin**
### What is changed and how it works?

Issue Number: Close https://github.com/tikv/tikv/issues/12240

What's Changed:

```commit-message
split the initializer from the endpoint
```

### Related changes

None

### Check List 

Tests 

- No code

Side effects

None

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by 5kbpers — COMMENTED** [review]
rest LGTM

File: components/cdc/src/initializer.rs [hunk 1]
```diff
@@ -0,0 +1,838 @@
...
+    // Fetched bytes to the scanner.
+    emit: usize,
+    // Bytes from the device, `None` if not possible to get it.
+    disk_read: Option<usize>,
+    // Perf delta for RocksDB.
+    perf_delta: PerfStatisticsDelta,
+}
+
+pub(crate) struct Initializer<E> {
+    pub(crate) engine: E,
```
[the remark this answers is not in the document]
Comment (hi-rustin) [reply components/cdc/src/initializer.rs:53]: I remember that rustc actually warns about using pub in pub(crate)(aka UNREACHABLE_PUB). But this warning is not turned on by default. 
It seems strange to use pub in pub(crate).
Also see: https://doc.rust-lang.org/stable/nightly-rustc/rustc_lint/builtin/static.UNREACHABLE_PUB.html
It also works for pub in pub(crate).

**Review by overvenus — APPROVED** [review]
LGTM :shipit:

**Review by 5kbpers — APPROVED** [review]

**Pull request merged**